src/base.jl

```diff
@@ -258,6 +258,58 @@ function mul!(x::AbstractVector, K::AbstractKroneckerProduct, v::AbstractVector)
     return x
 end
 
+"""
+    mul!(x::AbstractVector, K::AbstractKroneckerProduct, v::AbstractVector)
+
+Calculates the vector-matrix multiplication `K * v` and stores the result in
+`x`, overwriting its existing value.
+"""
+function mul!(x::AbstractVector, K::AbstractKroneckerProduct, v::AbstractVector)
+    M, N = getmatrices(K)
+    a, b = size(M)
+    c, d = size(N)
+    e = length(v)
+    f = length(x)
+    f == a * c || throw(DimensionMismatch(
+        "Dimension missmatch between kronecker system and result placeholder"))
+    e == b * d || throw(DimensionMismatch(
+        "Dimension missmatch between kronecker system and vector"))
+    if (d + a) * b < (b + c) * d
+        x .= vec(N * (reshape(v, d, b) * transpose(M)))
+    else
+        x .= vec((N * reshape(v, d, b)) * transpose(M))
+    end
+    return x
+end
+
+import Base.ReshapedArray
+
+function mul!(x::AbstractVector, K::AbstractKroneckerProduct,
+                                            v::ReshapedArray)
+    M, N = getmatrices(K)
+    a, b = size(M)
+    c, d = size(N)
+    e = length(v)
+    f = length(x)
+    f == a * c || throw(DimensionMismatch(
+        "Dimension missmatch between kronecker system and result placeholder"))
+    e == b * d || throw(DimensionMismatch(
+        "Dimension missmatch between kronecker system and vector"))
+
+    if size(v.parent) == (d, b)
+        V = v.parent
+    else
+        V = reshape(v, d, b)
+    end
+    if (d + a) * b < (b + c) * d
+        x .= vec(N * (V * transpose(M)))
+    else
+        x .= vec((N * V) * transpose(M))
```

Comment:
Make a function barrier for matrix multiplications?

Something like you suggested in https://github.com/MichielStock/Kronecker.jl/issues/14#issuecomment-515499247

```julia

function vecmulR!(X,N,V,M)
    X .= N * (V * transpose(M))
end

function vecmulL!(X,N,V,M)
    X .= (N * V) * transpose(M)
end

function vecmul!(X,N,V,M)
    X .= N * V * transpose(M)
end
```

You could allocate `X` in `mul!` and call
```julia
x .= vec(X)
```
Don't have to export, but a convenient function when `vec` is not required.

Reply:
Added these, and moved vec trick stuff to a new file (232b60e etc)